pyvlx/opening_device.py

```diff
@@ -182,23 +183,49 @@ async def set_position(self, position, wait_for_completion=True):
                 without stopping the blind (if orientation position has been changed.)
             * wait_for_completion: If set, function will return
                 after device has reached target position.
+            * orientation: If set, the orientation of the device will be set in the same request.
+                Note, that, if the position is set to 0, the orientation will be set to 0 too.
 
         """
         self.target_position = position
         self.position = position
 
+        kwargs = {}
+
+        if orientation is not None:
+            kwargs['fp3'] = orientation
+        elif self.target_position == Position(position_percent=0):
+            kwargs['fp3'] = Position(position_percent=0)
```

Comment:
I think this will not work. Line 197 wont be checked if the node has an orientation value, which should be the actual orientation of the blind. This should be changed to

```
        if self.target_position == Position(position_percent=0):
            kwargs['fp3'] = Position(position_percent=0)
        elif orientation is not None:
            kwargs['fp3'] = orientation
```
Moreover I would be careful with `kwargs['fp3'] = orientation`, because orientation is the current orientation of the blind, which can be updated via node_updater during blind movement. The question should be, in which case we should set  target_orientation = orientation. I would suggest if the blinds has finished its operations
`EXECUTION_COMPLETED = 0  # Execution is completed with no errors.`
we can use a valid orientation value from node_updater in order to update also the target_orientation.

Reply:
I'm not sure how to catch the following: Currently if I change the blind position via up, down and stop buttons in HA, I can return to previous orientation if I press two times the stop button. On the second stop command the blinds return to previous orientation. Not sure if the behaviour was like this from the beginning. However I like the function that I can lift or lower the blinds and return to previous orientation. Strange is only that I need to repeat the stop command for this. Now if we would set `target_orientation = orientation` and after first stop the `EXECUTION_COMPLETED = 0` would be submitted and also a valid orientation, I may could lose this functionality.

Reply:
I disagree. orientation is only ever set if it is called from stop function in which case it is set to target_orientation (note it is not self.position but the function parameter position) https://github.com/Julius2342/pyvlx/blob/e6ac8a4e5e1986db3f96fb4f0f4b54552abb6c3d/pyvlx/opening_device.py#L257 . I purposely chose this order to still be able to override the orientation if desired, but do admit it probably is an almost useless use case. Still I believe if the user wants to do it he should be able to.

I have never seen the two-times-top-button behavior on my blinds. Does this only happen if you do it shortly after each other?

Reply:
And of course, thanks for reviewing!

Reply:
> I disagree. orientation is only ever set if it is called from stop function in which case it is set to target_orientation (note it is not self.position but the function parameter position)
> 
But orientation is set every time you adjust the orientation, see [here](https://github.com/dumpfheimer/pyvlx/blob/e6ac8a4e5e1986db3f96fb4f0f4b54552abb6c3d/pyvlx/opening_device.py#L273) 
and also every time you receive a valid information, see [here](https://github.com/dumpfheimer/pyvlx/blob/e6ac8a4e5e1986db3f96fb4f0f4b54552abb6c3d/pyvlx/node_updater.py#L35).

Reply:
> I have never seen the two-times-top-button behavior on my blinds. Does this only happen if you do it shortly after each other?

On first stop command the blinds stop immediately their movement (and the orientation is always closed during position change), on the second stop command the orientations return to previous orientation.

Reply:
> > I disagree. orientation is only ever set if it is called from stop function in which case it is set to target_orientation (note it is not self.position but the function parameter position)
> 
> But orientation is set every time you adjust the orientation, see [here](https://github.com/dumpfheimer/pyvlx/blob/e6ac8a4e5e1986db3f96fb4f0f4b54552abb6c3d/pyvlx/opening_device.py#L273) and also every time you receive a valid information from KLF200, see [here](https://github.com/dumpfheimer/pyvlx/blob/e6ac8a4e5e1986db3f96fb4f0f4b54552abb6c3d/pyvlx/node_updater.py#L35).

those are both examples of setting the self.orientation field which is not what is checked in the condition, it is referencing the function parameter called orienation (it is not self.orientation). And that function parameter is only not None if it is being called by the stop function.

Do you see what I mean?

Reply:
![image](https://user-images.githubusercontent.com/60989727/142268241-541e89f5-aad7-4429-ac0c-b78ecae48e2d.png)

Sorry, I am having difficulties expressing myself in a different way. Maybe this image can halp clarify.

Reply:
Oh yes, sorry for this, my fault... the missing `self.`

Reply:
Yes, exactly. Do you still see any issues with the code?

Reply:
No, I've no problems with this and I have already finished the review and approved your changes. I also switched my custom to test your pyvlx modification and till now everthing works fine!
We can maybe think about a good solution to adjust node.target_orientation and node.target_position, because currently they wont be changed if you use your remotes and not HA. However this could be done within a seperate PR.

Reply:
Nice, thanks!
In my opinion target_position and target_orientation do not have to be changed by anyone other than HA because they are only used when they are changed, which can only happen from HA. Am I missing something?

Reply:
Basically target is what the user wan't to achieve.  In HA you may want to change the position, but maybe you set prevously the orientation with your remote and want to keep this orientation. However HA would only use the last orientation set via HA, because it is stored as target orientation!

Reply:
Unless I messed up something, in that case the Ignore parameter value should be sent, which would not override the position/orientation set by the remote. Furthermore the position should be updated by the HomeStatusMonitor (or so) and the orientation should be updated by the new code of @Nicks57 which is part of this PR. Is this not working for you? Maybe I missed something.